statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,257 @@
+
+
+    def mean(self, method):
+        """
+        Calculates survey mean
+
+        Parameters
+        ----------
+        self
+
+        Returns
+        -------
+        A tuple of length 2, each index is a 1xp array. The first array holds 
+        the mean for each of the p vars in self.data. The second array is the
+        SE for each total 
+        """
+        try:
+            self._mean = np.round(self._total / np.sum(self.prob_weights), 2)
+        # if self.total doesn't exist yet
+        except AttributeError:
+            self._total = self.total()
+            self._mean = self._total / np.sum(self.prob_weights)
+
+        if method == "jack":
+            mean_se = np.array([self._jackknife("mean", index, SurveyMean(self.data, self.cluster, self.strata, self.prob_weights,index, self._mean)) for index in range(self.data.shape[1])])
+        
+        return self._mean, mean_se
+
+    # need to make this work when input is an array-like object
+    def percentile(self, percentile): 
+        p = self.data.shape[1]
+        self._percentile = np.empty(p)
+        
+        for index in range(p):
+            sorted_weights = [x for (y,x) in sorted(zip(self.data[:, index],self.prob_weights))]
+            sorted_weights = np.array(sorted_weights)
+            cumsum_weights = np.cumsum(sorted_weights)
+
+            perc = (cumsum_weights[-1] * percentile) / 100
+            sorted_data = np.sort(self.data[:, index])
+            
+            if perc in cumsum_weights:
+                ind = np.where(cumsum_weights == perc)[0].item()
+                self._percentile[index] = (sorted_data[ind] + sorted_data[ind + 1]) / 2
+            else:
+                ind = np.argmax(cumsum_weights > perc)
+                self._percentile[index] = sorted_data[ind]
+        return self._percentile
+
+    def median(self):
+        self._median = self.percentile(50)
+        return self._median
+
+
+class SurveyMean(SurveyStat):
+
+    def __init__(self, data, cluster, strata, prob_weights, column_index,parent_stat):
+        SurveyStat.__init__(self, data, cluster, strata, prob_weights)
+        self.column_index = column_index
+        self.parent_stat = parent_stat
+
+    def _stat(self, new_weights):
+        stat = np.array(np.dot(self.data[:, self.column_index], new_weights).item())
+        stat /= np.sum(new_weights)
+        stat -= self.parent_stat[self.column_index]
+        return stat
+
+class SurveyTotal(SurveyStat):
+
+    def __init__(self, data, cluster, strata, prob_weights, column_index, parent_stat):
+        SurveyStat.__init__(self, data, cluster, strata, prob_weights)
+        self.column_index = column_index
+        self.parent_stat = parent_stat
+
+    def _stat(self, new_weights):
+        stat = np.array(np.dot(self.data[:, self.column_index], new_weights).item())
+        stat -= self.parent_stat[self.column_index]
+        return stat
+
+
+
+
+df = pd.read_csv("~/Documents/survey_data")
```

Comment:
Try to avoid committing code like this (from here to end).  You can have local files for running quick checks while developing that are never added to the repo.